components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -686,6 +706,93 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         return consentPage;
     }
 
+    private static String getFilteredScopes(OAuth2Parameters params) throws OAuthSystemException {
+
+        try {
+            Set<String> allowedScopes = params.getScopes();
+            startTenantFlow(params.getTenantDomain());
+
+            /* If DropUnregisteredScopes scopes config is enabled
+             then any unregistered scopes(excluding internal scopes
+             and allowed scopes) is be dropped. Therefore they will
+             not be shown in the user consent screen.*/
+            if (oauthServerConfiguration.isDropUnregisteredScopes()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+                }
+                allowedScopes = dropUnregisteredScopes(params);
+            }
+
+            // Get registered OIDC scopes.
+            String[] oidcScopes = oAuthAdminService.getScopeNames();
+            List<String> oidcScopeList = new ArrayList<>(Arrays.asList(oidcScopes));
+
+            /* Remove OIDC scopes from scope string sent back
+             in the consent page as consent is not needed for
+             OIDC scopes.*/
+            StringBuilder scopes = new StringBuilder();
+            for (String scope : allowedScopes) {
```

Comment:
Is this valid for all OIDC scope or just the "openid" scope?

Reply:
All oidc scopes. Since OIDC scopes are for claim requests we will show the requested claims and remove it from showing under permission scopes

Reply:
+1